staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/named_certificates_test.go

```diff
@@ -0,0 +1,302 @@
+package dynamiccertificates
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type testCertSpec struct {
+	host       string
+	names, ips []string // in certificate
+}
+
+type namedtestCertSpec struct {
+	testCertSpec
+	explicitNames []string // as --tls-sni-cert-key explicit names
+}
+
+func TestBuiltNamedCertificates(t *testing.T) {
```

Comment:
Simply moved from `staging/src/k8s.io/apiserver/pkg/server/options/serving_test.go` and updated to build cert providers